Make `PresentMode` configurable.

Yatekii opened the pull request.

kvark commented on wgpu-native/src/device.rs:
```diff
@@ -1731,7 +1731,7 @@ pub fn device_create_swap_chain(
         conv::map_texture_format(desc.format),
         num_frames, //TODO: configure?
-    );
+    ).with_mode(desc.present_mode);
```
let's just do it like `composite_alpha` - with a simple assignment

kvark commented on wgpu-native/src/swap_chain.rs:
```diff
@@ -77,13 +77,30 @@ pub struct SwapChain<B: hal::Backend> {
 }
 
+#[repr(C)]
+#[derive(Clone, Debug)]
```
let's also derive `Copy`

kvark commented on wgpu-native/src/device.rs:
```diff
@@ -1734,6 +1734,7 @@ pub fn device_create_swap_chain(
     );
     //TODO: check for supported
     config.composite_alpha = hal::window::CompositeAlpha::OPAQUE;
+    config.present_mode = desc.present_mode.clone().into();
```
let's just do a `match` here without implementing `into()`

Yatekii commented:
The 

> Thank you for addressing the concerns!
> Where did wgpu.h go?

Well I can't build the C examples as of now still ... I just made an intermediate commit as it builds for Rust only.
Dunno how to fix it.
But ofc that gets pushed to the PR.

kvark commented on wgpu-native/src/device.rs:
```diff
@@ -1734,6 +1734,10 @@ pub fn device_create_swap_chain(
     );
     //TODO: check for supported
     config.composite_alpha = hal::window::CompositeAlpha::OPAQUE;
+    config.present_mode = match &desc.present_mode {
```
shouldn't need the "&"

kvark replied:
this is still not fixed?

kvark commented:
bors r+

kvark commented:
bors try

kvark commented:
Bors r+

> On Jul 16, 2019, at 19:41, bors[bot] <notifications@github.com> wrote:
> 
> try
> 
> Build succeeded
> 
> continuous-integration/travis-ci/push
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub, or mute the thread.
